src/irmin-pack/unix/io.ml

```diff
@@ -0,0 +1,282 @@
+(*
+ * Copyright (c) 2022-2022 Tarides <contact@tarides.com>
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ *)
+
+open! Import
+open Io_intf
+
+(* File utils, taken from index.unix package *)
+module Util = struct
+  module Syscalls = Index_unix.Syscalls
+
+  (* NOTE following copied from Index_unix.Raw *)
```

Comment:
can we maybe use the Raw API instead of copying these functions?

Reply:
The `really` functions are not exported in raw

Reply:
I know, I meant using the functions exposed in the API (`unsafe_read` and `unsafe_write`). We would also have the stats for those. But ok anyway.